docs/guide/en/field-configuration.md

```diff
@@ -1,33 +1,150 @@
-# Fields configuration
+# Field configuration
 
 Fields could be configured on multiple levels:
 
-- widget factory configuration;
-- common configuration from theme (optional);
-- individual configuration from theme (optional);
-- configuration at the place of use.
-
-More specific configuration has more priority.
+- Widget factory configuration:
+  - With definitions;
+  - With themes and default theme selected;
+  - With themes and theme specified at field level.
+- Widget configuration.
 
 ## Widget factory configuration
 
 Widget factory could be used to configure fields. It is handy to use it to set global defaults.
 
-TODO: provide example
+### With definitions
+
+You can use definitions:
+
+```php
+use Psr\Container\ContainerInterface;
+use Yiisoft\Form\Field\Base\BaseField;
+use Yiisoft\Form\Field\Base\InputField;use Yiisoft\Form\Field\Base\PartsField;use Yiisoft\Form\Field\Text;
+use Yiisoft\Widget\Widget;
+use Yiisoft\Widget\WidgetFactory;
+
+WidgetFactory::initialize(
+    /** @var ContainerInterface $container */
+    $container,
+    definitions: [        
+        Text::class => [
+            'containerClass()' => ['field-container']
+        ],
+    ], 
+);
+echo Text::widget();
+```
+
+Result:
+
+```html
+<div class="field-container">
+    <input type="text">
+</div>
+```
+
+### With themes
+
+Alternatively, themes can be used. To apply the theme automatically, set default theme.
```

Comment:
```suggestion
Themes could be used in case you want to have multiple configuration sets
and the ability to switch from one to another. To apply the theme automatically,
specify it as default theme.
```